t/unit/app/test_utils.py

```diff
@@ -26,6 +26,13 @@ def test_find_value_for_key(self):
         assert self.app.conf.find_value_for_key(
             'always_eager') is False
 
+    def test_broker_priority(self, monkeypatch):
+        prepatch_broker_url = 'memory://thisshouldbethebroker'
+        self.app.conf.broker_url = prepatch_broker_url
+        monkeypatch.setenv('CELERY_BROKER_URL',
+                           'memory://thisshouldntbethebroker')
+        assert self.app.conf.broker_url == prepatch_broker_url
```

Comment:
The application configuration has already been resolved at this stage, so I think this test case does not actually prove the issue. Configuration cannot be changed dynamically.

Reply:
I got that feeling when I wrote this, however calling `self.app.conf.broker_url` after setting it to `prepatch_broker_url`, but before setting `CELERY_BROKER_URL` in the environment, it does return the `prepatch_broker_url` string